enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,366 @@
+
+In addition, by dynamically learning routes and programming them into OVN, we risk a customer accidentally introducing
+potentially hundreds or even thousands of routes into different OVN logical routers. As a mitigation, we can recommend
+using communities or route aggregation to customers to limit RIB size. We will also need to scale test the effect of
+many routes inside OVN logical routers.
+
+Also, FRR-K8S is not GA yet, and targeted for 4.17.
+
+### Drawbacks
+
+Increased complexity of our SDN networking solution to support more complex networking. Additionally, it increases
+support complexity, because we are peering with a user’s fabric now and no longer blindly tunneling through their
+network with Geneve.
+
+### Implementation Details/Notes/Constraints
+
+#### BGP Peering
+
+FRR-K8S shall support both internal BGP (iBGP) and external BGP (eBGP). With iBGP, a full mesh topology is required. This
+can be cumbersome and have issues with scaling in large clusters. Therefore, FRR-K8S shall also support designating one
+or more nodes as route reflectors. With route reflectors, nodes only have to connect to the route reflectors and do not
+have to maintain a full mesh with all other nodes in the cluster.
+
+#### FRR-K8S Integration
+
+As previously mentioned frr-k8s will be used in order to deploy and manage FRR configuration. Support will be added
+where necessary for gaps in frr-k8s API, but in the meantime frr-k8s supports also issuing raw FRR configuration. The
+raw configuration will be for things that OVN-Kubernetes needs to configure, and not something exposed to a user. A user
+will need the ability to:
+
+* Define BGP peers, ASN, along with other common BGP configurations like communities, route filtering, etc.
+* Configure BFD
+
+For standard BGP support with the default VRF, frr-k8s API can be used in order to configure everything. In this case a
+user will create frr-k8s CRs for their desired BGP configuration. OVN-Kubernetes will watch for these CRs, and once
+created, OVN-Kubernetes will start monitoring the kernel routing table via netlink for routes installed by FRR. When FRR
+is using Quagga, the routes can be identified via a proto label:
+
+```
+192.168.1.0/24 via 10.0.0.1 dev eth0 proto zebra
+```
+
+When these routes are seen on the node, ovnkube-controller will update the routes in NBDB for any GW routers on the
+node, filtering the route for the GW router based on if the interface on the route matches the interface in OVS. This
+will happen irrespective of gateway mode, as some features in local gateway mode rely on the GRs for sending traffic
+(Egress IP, MEG). Note, this configuration will only happen on nodes selected by the frr-k8s CRs.
+
+#### API Changes
+
+##### Route Advertisements
+
+When OVN-Kubernetes detects that a FRR-K8S CR has been created for BGP peering, OVN-Kubernetes will by default
+advertise the pod subnet for each node, via creating an additive frr-k8s CR that is managed by OVN-Kubernetes.
```

Comment:
an additive configuration per node* as each node will have a separate slice of that network CIDR.